### ISSUE TITLE
Poor error message when Index Checker fails to identify required @LTOMLengthOf(String[] names) or @LTLengthOf(value = “”, offset = “” )

### ISSUE THREAD

**Issue by NhatDinh** [typetools/checker-framework#1902]
Consider:
```java
public class SimpleMethod3 {
   @NonNegative int size;
   int[] vDown = new int[size];
   //simpler version
   void method3(@NonNegative int size,@NonNegative int value){
       this.size = size;
       this.vDown = new int[this.size];
       vDown[1 + value] = 10;
   }
   public static void main(String args[]) {
   }
}
```

Received warnings:
```java
Error:(14, 17) java: [array.access.unsafe.high.range] Potentially unsafe array access: the index could be larger than the array's bound
  index type found: @IntRange(from=-2147483648, to=2147483647) int
  array type found: @UnknownVal int @UnknownVal []
  required        : index of type @IndexFor("this.vDown") or @LTLengthOf("this.vDown"), or array of type @MinLen("2147483648")
```
None of the prompted required type would work. On the other hand,  @LTLengthOf(value = "this.vDown", offset = "1") or @LTOMLengthOf(“this.vDown”) will pass the checker. Also, @MinLen("2147483648") don't make much sense to me, why does array need to have minimum of 2147483648 element? I think it should be removed. To be more precise, the checker should prompt:
```java
 required: index of type@LTLengthOf(value = "this.vDown", offset = “”), or @LTOMLengthOf(“this.vDown”)
```
If this is a valid concern, I would like to work on it.

**Comment by kelloggm** [comment]
Sorry for taking so long to get back to you.

I agree that the error message in this case is not very helpful. The Index Checker uses a heuristic to decide what kind of error message to prompt for, and prefers not to prompt for error messages involving an offset when it can. But it could be better! We'd appreciate help coming up with a way to issue a better error message in cases like this. The place to look in the code to see where we decide which error message to issue is in `UpperBoundVisitor#visitAccess`, around line 100.

**Comment by NhatDinh** [comment]
@kelloggm I'm working on this.
